src/installer/iohandler.py

```diff
@@ -0,0 +1,100 @@
+"""Handling wheel installation I/O-related functionality."""
+
+import hashlib
+import os
+from io import RawIOBase
+
+from installer._compat.typing import TYPE_CHECKING
+
+from .records import Hash, Record
+
+if TYPE_CHECKING:
+    from ._compat.typing import FSPath
+    from typing import Dict, Tuple, Optional, Callable
+
+_BLOCK_SIZE = 4096
+_SCHEME_KEYS = ["platlib", "purelib"]
+
+
+def basic_copy_handler(
+    source,  # type: RawIOBase
+    dest,  # type: FSPath
+    hash_algorithm="sha256",  # type: Optional[str]
```

Comment:
This shouldn't be `Optional`.

Reply:
What if the files is not supposed to have an hash? Which is the case of `RECORD`.